### PR TITLE
Pulsar IO: allow Sinks to use native AVRO and JSON

### PR DESCRIPTION
**Pull request by eolivelli**
The Sink should be able to manage the result of GenericRecord.getNativeObject().
In order to do this Apache AVRO must be loaded from the same classloader that is loading Apache Pulsar Runtime.

The same problem applies in the case of a JsonNode returned by getNativeObject();

*Modifications*
- Add AVRO to the list of classes (like slf4j) to be loaded from the Pulsar runtime (this in turn imports Commons Compress and Jackson Databind).
- Enhance the existing integration tests, that tested about the Schema definition, but it didn't actually try to "use" the `org.apache.avro.GenericRecord` object as well as the `Jackson JsonNode` object

### CODE REVIEW

File: pulsar-functions/runtime-all/src/test/java/org/apache/pulsar/functions/instance/JavaInstanceDepsTest.java [hunk 1]
```diff
@@ -60,11 +60,14 @@ public void testInstanceJarDeps() throws IOException {
             if (e == null)
                 break;
             String name = e.getName();
-            if (name.endsWith(".class") && !name.startsWith("META-INF")) {
+            if (name.endsWith(".class") && !name.startsWith("META-INF") && !name.equals("module-info.class")) {
                 // The only classes in the java-instance.jar should be org.apache.pulsar, slf4j, and log4j classes
```
Comment (nlu90) [inline pulsar-functions/runtime-all/src/test/java/org/apache/pulsar/functions/instance/JavaInstanceDepsTest.java:64]: the comments also need to be updated

**Comment by Anonymitaet** [comment]
Thanks for your contribution. For this PR, do we need to update docs?

(The [PR template contains info about doc](https://github.com/apache/pulsar/blob/master/.github/PULL_REQUEST_TEMPLATE.md#documentation), which helps others know more about the changes. Can you provide doc info in future PR descriptions? Thanks)

**Comment by eolivelli** [comment]
@Anonymitaet when we will find a good solution for this problem I will be happy to help with the docs.
This PR is part of a bigger issue with have with Pulsar IO and Pulsar Functions

currently this is only a draft

File: pulsar-functions/runtime-all/src/test/java/org/apache/pulsar/functions/instance/JavaInstanceDepsTest.java (hunk 1 above)
Comment (eolivelli) [reply pulsar-functions/runtime-all/src/test/java/org/apache/pulsar/functions/instance/JavaInstanceDepsTest.java:67]: done, thanks

**Review by sijie — APPROVED** [review]

**Comment by eolivelli** [comment]
@nlu90 can you please take a final look ?
I would like to fix this problem ASAP in 2.9.0 and commit this patch, this way @merlimat can move forward with his work about the classpath

**Review by nlu90 — APPROVED** [review]
👍🏼

**Pull request merged**